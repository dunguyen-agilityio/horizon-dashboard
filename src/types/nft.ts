// Models
import { NFT } from '@/models/NFT';

export interface INFTResponse {
  id: number;
  attributes: Omit<NFT, 'id'>;
}
