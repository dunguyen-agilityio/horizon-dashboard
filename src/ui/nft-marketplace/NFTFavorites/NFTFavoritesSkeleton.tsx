import NFTCardSkeleton from '../NFTCard/NFTCardSkeleton';

const NFTFavoritesSkeleton = () => (
  <div className="flex justify-center mt-1 sm:mt-4">
    <div className="flex flex-wrap justify-center gap-5 sm:justify-start max-w-[738px] xl:max-w-[1084px] 2xl:max-w-[1500px] w-full">
      {Array(8)
        .fill(0)
        .map((_, index) => (
          <NFTCardSkeleton key={index} />
        ))}
    </div>
  </div>
);

export default NFTFavoritesSkeleton;
