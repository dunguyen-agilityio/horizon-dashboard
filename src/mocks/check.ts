import { TCheck } from '@/types/check';

export const MOCK_CHECKS: TCheck[] = [
  {
    id: '1',
    name: 'Tony Reichert',
    progress: 45,
    quantity: 20,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '2',
    name: 'Zoey Lang',
    progress: 15,
    quantity: 20,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '3',
    name: 'Jane Fisher',
    progress: 55,
    quantity: 28,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '4',
    name: 'William Howard',
    progress: 65,
    quantity: 30,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
];
