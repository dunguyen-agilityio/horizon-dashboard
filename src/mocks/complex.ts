import { Complex, ComplexStatus } from '@/types/complex';

export const MOCK_COMPLEXES: Complex[] = [
  {
    id: '1',
    name: 'Tony Reichert',
    progress: 45,
    status: ComplexStatus.APPROVED,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '2',
    name: 'Zoey Lang',
    progress: 15,
    status: ComplexStatus.DISABLE,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '3',
    name: 'Jane Fisher',
    progress: 55,
    status: ComplexStatus.ERROR,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '4',
    name: 'William Howard',
    progress: 65,
    status: ComplexStatus.APPROVED,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
];
