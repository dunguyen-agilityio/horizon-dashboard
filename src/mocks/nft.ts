// Types
import { NFTResponse, type NFTData } from '@/models/NFT';

// Constants
import { NFT_IMAGES } from '@/constants/images';

// Mocks
import { MOCK_IMAGE_RESPONSE, MOCK_USERS, MOCK_USERS_RESPONSE } from './user';
import { StrapiResponse } from '@/types/strapi';

const [MOCK_USER] = MOCK_USERS;

export const MOCK_NFTS: NFTData[] = [
  {
    id: '1',
    name: 'Mesh Gradients',
    author: MOCK_USER,
    price: 1.23,
    image: { url: NFT_IMAGES.DEFAULT, hash: 'mock hash' },
    members: MOCK_USERS,
    createdAt: new Date('2024-08-17T07:25:04.188Z'),
  },
];

export const MOCK_NFTS_RESPONSE: StrapiResponse<NFTResponse>[] = [
  {
    id: '1',
    attributes: {
      name: 'Mesh Gradients',
      author: { data: MOCK_USERS_RESPONSE[0] },
      price: 1.23,
      members: { data: MOCK_USERS_RESPONSE },
      image: { data: MOCK_IMAGE_RESPONSE },
      createdAt: new Date('2024-08-17T07:25:04.188Z'),
    },
  },
];
