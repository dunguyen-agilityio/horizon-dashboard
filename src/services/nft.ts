import { NFTResponse } from '@/models/NFT';
import { User } from '@/models/User';
import { apiClient } from './api';
import { API_ENTITY } from '@/constants';
import { StrapiModelWithPagination, StrapiResponse } from '@/types/strapi';

type UserFavoriteResponse = User & {
  favorite: {
    id: string;
    nfts: NFTResponse[];
  } | null;
};

export const getNFTsFavorite = async () => {
  const searchParams = new URLSearchParams({
    '[populate][favorite][populate][nfts][populate][image]': 'url',
    '[populate][favorite][populate][nfts][populate][members][populate][avatar]':
      'url',
    '[populate][favorite][populate][nfts][populate][author][populate][avatar]':
      'url',
  });

  const { data, error } = await apiClient.get<UserFavoriteResponse>(
    `${API_ENTITY.USERS}/me?${searchParams.toString()}`,
    { cache: 'no-store' },
  );

  return { data: data?.favorite?.nfts || [], error };
};

type NFTsResponse = StrapiModelWithPagination<StrapiResponse<NFTResponse>[]>;

export const getNFTsRecentlyAdded = async () => {
  const searchParams = new URLSearchParams({
    'sort[0]': 'createdAt:desc',
    '[populate][image]': '*',
    '[populate][members][populate][avatar]': '*',
    '[populate][author][populate][avatar]': '*',
    '[pagination][page]': '1',
    '[pagination][pageSize]': '6',
  });

  const { data, error } = await apiClient.get<NFTsResponse>(
    `${API_ENTITY.NFTS}?${searchParams.toString()}`,
    { cache: 'no-store' },
  );

  return { data: data?.data || [], error };
};

export const getNFTsHistory = async () => {
  const searchParams = new URLSearchParams({
    'sort[0]': 'createdAt:asc',
    '[populate][image]': '*',
    '[populate][members][populate][avatar]': '*',
    '[populate][author][populate][avatar]': '*',
    '[pagination][page]': '1',
    '[pagination][pageSize]': '6',
  });

  const { data, error } = await apiClient.get<NFTsResponse>(
    `${API_ENTITY.NFTS}?${searchParams.toString()}`,
    { cache: 'no-store' },
  );

  return { data: data?.data || [], error };
};
