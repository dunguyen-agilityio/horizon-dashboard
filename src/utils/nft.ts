import { NFTDataExtend, NFTResponse } from '@/models/NFT';
import { StrapiModel, StrapiResponse } from '@/types/strapi';
import { TUser, User } from '@/models/User';
import { TImage } from '@/models/Image';

export const formatStrapiModel = <T extends { id: string }>({
  attributes,
  id,
}: StrapiResponse<T>) =>
  ({
    ...attributes,
    id,
  }) as T;

export type UserResponse1 = Omit<TUser, 'avatar' | 'role'> & {
  avatar: StrapiModel<StrapiResponse<TImage>> | null;
  role: StrapiResponse<{ name: string; id: string }> | null;
};

const formatUserResponse = (data: StrapiResponse<UserResponse1>): User => {
  const { avatar, role, ...rest } = formatStrapiModel(data);

  const { attributes: avatarData, id } = avatar?.data ?? {};

  const hasAvatar = avatarData && id;

  return new User({
    ...rest,
    role: { name: role?.attributes.name ?? '' },
    ...(hasAvatar && {
      avatar: { hash: avatarData.hash, url: avatarData.url },
    }),
  });
};

export const formatNFTResponse = ({
  id,
  attributes: { author, image, members, ...rest },
}: StrapiResponse<NFTResponse>): NFTDataExtend => {
  const { hash = '', url = '' } = image?.data?.attributes || {};

  return {
    id,
    ...rest,
    author: formatUserResponse(author.data),
    members: members.data?.map?.((member) => formatUserResponse(member)) || [],
    image: { hash, url },
  };
};
