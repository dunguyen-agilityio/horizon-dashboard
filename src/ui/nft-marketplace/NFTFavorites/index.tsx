'use client';

import { useState } from 'react';

// Components
import NFTCard from '../NFTCard';
import { Pagination } from '@nextui-org/pagination';
import { ItemNotFound } from '@/components';

// Constants
import { PUBLIC_ROUTES } from '@/constants';

// Models
import { NFTResponse } from '@/models/NFT';

// Types
import { StrapiResponse } from '@/types/strapi';

// Utils
import { formatNFTResponse } from '@/utils/nft';

interface INFTFavoriteProps {
  listFavorites: StrapiResponse<NFTResponse>[];
  pageSize: number;
  activePage: number;
}

const NFTFavorites = ({
  listFavorites,
  pageSize,
  activePage,
}: INFTFavoriteProps) => {
  const [currentPage, setCurrentPage] = useState<number>(activePage);

  if (listFavorites.length === 0)
    return (
      <ItemNotFound
        title="Start Find Your Favorite NFTs"
        description="Browse our collection and click the heart icon to add NFTs to your favorites list!"
        href={PUBLIC_ROUTES.NFT_MARKETPLACE}
        label="Explore NFTs"
      />
    );

  return (
    <div className="flex justify-center mt-1 sm:mt-4">
      <div className="flex flex-wrap justify-center gap-5 sm:justify-start max-w-[738px] xl:max-w-[1084px] 2xl:max-w-[1500px] w-full">
        {listFavorites.map((nft) => {
          const { members, author, ...favorite } = formatNFTResponse(nft);
          return (
            <NFTCard
              key={favorite.id}
              data-testid={`nft-card-${favorite.id}`}
              isShowIcon={false}
              author={author}
              members={members}
              {...favorite}
            />
          );
        })}
        {!!listFavorites.length && (
          <div className="w-full flex justify-center lg:justify-end my-8">
            <Pagination
              data-testid="pagination"
              showControls
              showShadow
              color="primary"
              total={pageSize}
              page={currentPage}
              onChange={setCurrentPage}
            />
          </div>
        )}
      </div>
    </div>
  );
};

export default NFTFavorites;
