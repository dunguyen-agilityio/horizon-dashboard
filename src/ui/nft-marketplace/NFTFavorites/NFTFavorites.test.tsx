import { render } from '@testing-library/react';

// Component
import NFTFavorites from '.';

// Mocks
import { MOCK_NFTS } from '@/mocks/nft';

describe('NTFFavorites', () => {
  it('should match snapshot', () => {
    const container = render(
      <NFTFavorites NFTFavoriteData={MOCK_NFTS} pageSize={5} page={1} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correct info of NFT card', () => {
    const { getByText } = render(
      <NFTFavorites NFTFavoriteData={MOCK_NFTS} pageSize={5} page={1} />,
    );
    expect(getByText('Mesh Gradients')).toBeInTheDocument();
    expect(getByText('By Adela Parkson')).toBeInTheDocument();
  });

  it('should render the correct number of NFT cards', () => {
    const { getAllByTestId } = render(
      <NFTFavorites NFTFavoriteData={MOCK_NFTS} pageSize={5} page={1} />,
    );

    const nftCards = getAllByTestId('nft-card');
    expect(nftCards).toHaveLength(MOCK_NFTS.length);
  });
});
