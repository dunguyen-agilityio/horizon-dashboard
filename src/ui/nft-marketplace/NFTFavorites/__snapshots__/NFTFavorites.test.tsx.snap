// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NTFFavorites should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex justify-center mt-1 sm:mt-4"
      >
        <div
          class="flex flex-wrap justify-center gap-5 sm:justify-start max-w-[738px] xl:max-w-[1084px] 2xl:max-w-[1500px] w-full"
        >
          <a
            class="relative"
            data-testid="nft-card"
            href="/nft-marketplace/1"
          >
            <div
              class="bg-white dark:bg-indigo rounded-md p-5"
            >
              <div
                class="relative w-[308px] h-[205px] rounded-[18px] overflow-hidden"
              >
                <img
                  alt="Mesh Gradients"
                  class="object-cover"
                  data-nimg="fill"
                  decoding="async"
                  loading="lazy"
                  sizes="100vw"
                  src="/_next/image?url=%2Fnft-default.webp&w=3840&q=75"
                  srcset="/_next/image?url=%2Fnft-default.webp&w=640&q=75 640w, /_next/image?url=%2Fnft-default.webp&w=750&q=75 750w, /_next/image?url=%2Fnft-default.webp&w=828&q=75 828w, /_next/image?url=%2Fnft-default.webp&w=1080&q=75 1080w, /_next/image?url=%2Fnft-default.webp&w=1200&q=75 1200w, /_next/image?url=%2Fnft-default.webp&w=1920&q=75 1920w, /_next/image?url=%2Fnft-default.webp&w=2048&q=75 2048w, /_next/image?url=%2Fnft-default.webp&w=3840&q=75 3840w"
                  style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
                />
              </div>
              <div
                class="flex justify-between pt-5"
              >
                <div>
                  <p
                    class="text-primary dark:text-white text-extra font-bold leading-large text-large"
                  >
                    Mesh Gradients
                  </p>
                  <p
                    class="text-secondary dark:text-secondary text-small font-medium"
                  >
                    By User 1
                  </p>
                </div>
                <div
                  class="flex items-center justify-center h-auto w-max [&>span]:w-[27px] [&>span]:h-[27px] [&>span]:ring-offset-0"
                  role="group"
                >
                  <span
                    class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-10 h-10 text-tiny bg-default text-default-foreground rounded-full ring-2 ring-offset-2 ring-offset-background dark:ring-offset-background-dark data-[hover=true]:-translate-x-3 rtl:data-[hover=true]:translate-x-3 transition-transform data-[focus-visible=true]:-translate-x-3 rtl:data-[focus-visible=true]:translate-x-3 ring-default ms-0"
                    tabindex="-1"
                  >
                    <img
                      alt="Adela Parkson"
                      class="flex object-cover w-full h-full transition-opacity !duration-500 opacity-0 data-[loaded=true]:opacity-100"
                      data-nimg="1"
                      decoding="async"
                      height="27"
                      loading="lazy"
                      src="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=64&q=75"
                      srcset="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=64&q=75 2x"
                      style="color: transparent;"
                      width="27"
                    />
                  </span>
                  <span
                    class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-10 h-10 text-tiny bg-default text-default-foreground rounded-full ring-2 ring-offset-2 ring-offset-background dark:ring-offset-background-dark data-[hover=true]:-translate-x-3 rtl:data-[hover=true]:translate-x-3 transition-transform data-[focus-visible=true]:-translate-x-3 rtl:data-[focus-visible=true]:translate-x-3 ring-default -ms-2"
                    tabindex="-1"
                  >
                    <img
                      alt="user2"
                      class="flex object-cover w-full h-full transition-opacity !duration-500 opacity-0 data-[loaded=true]:opacity-100"
                      data-nimg="1"
                      decoding="async"
                      height="27"
                      loading="lazy"
                      src="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=64&q=75"
                      srcset="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=64&q=75 2x"
                      style="color: transparent;"
                      width="27"
                    />
                  </span>
                  <span
                    class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-10 h-10 text-tiny bg-default text-default-foreground rounded-full ring-2 ring-offset-2 ring-offset-background dark:ring-offset-background-dark data-[hover=true]:-translate-x-3 rtl:data-[hover=true]:translate-x-3 transition-transform data-[focus-visible=true]:-translate-x-3 rtl:data-[focus-visible=true]:translate-x-3 ring-default -ms-2"
                    tabindex="-1"
                  >
                    <img
                      alt="user3"
                      class="flex object-cover w-full h-full transition-opacity !duration-500 opacity-0 data-[loaded=true]:opacity-100"
                      data-nimg="1"
                      decoding="async"
                      height="27"
                      loading="lazy"
                      src="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=64&q=75"
                      srcset="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=64&q=75 2x"
                      style="color: transparent;"
                      width="27"
                    />
                  </span>
                  <span
                    class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-10 h-10 text-tiny bg-default text-default-foreground rounded-full ring-2 ring-offset-2 ring-offset-background dark:ring-offset-background-dark -ms-2 data-[hover=true]:-translate-x-3 rtl:data-[hover=true]:translate-x-3 transition-transform data-[focus-visible=true]:-translate-x-3 rtl:data-[focus-visible=true]:translate-x-3 ring-default hover:-translate-x-0"
                    tabindex="-1"
                  >
                    <span
                      aria-label="+1"
                      class="absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 font-normal text-center text-inherit"
                      role="img"
                    >
                      +1
                    </span>
                  </span>
                </div>
              </div>
              <p
                class="text-primary dark:text-white text-small text-blue-450 font-bold pt-[25px]"
              >
                Current Bid: 1.23 USD
              </p>
            </div>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex justify-center mt-1 sm:mt-4"
    >
      <div
        class="flex flex-wrap justify-center gap-5 sm:justify-start max-w-[738px] xl:max-w-[1084px] 2xl:max-w-[1500px] w-full"
      >
        <a
          class="relative"
          data-testid="nft-card"
          href="/nft-marketplace/1"
        >
          <div
            class="bg-white dark:bg-indigo rounded-md p-5"
          >
            <div
              class="relative w-[308px] h-[205px] rounded-[18px] overflow-hidden"
            >
              <img
                alt="Mesh Gradients"
                class="object-cover"
                data-nimg="fill"
                decoding="async"
                loading="lazy"
                sizes="100vw"
                src="/_next/image?url=%2Fnft-default.webp&w=3840&q=75"
                srcset="/_next/image?url=%2Fnft-default.webp&w=640&q=75 640w, /_next/image?url=%2Fnft-default.webp&w=750&q=75 750w, /_next/image?url=%2Fnft-default.webp&w=828&q=75 828w, /_next/image?url=%2Fnft-default.webp&w=1080&q=75 1080w, /_next/image?url=%2Fnft-default.webp&w=1200&q=75 1200w, /_next/image?url=%2Fnft-default.webp&w=1920&q=75 1920w, /_next/image?url=%2Fnft-default.webp&w=2048&q=75 2048w, /_next/image?url=%2Fnft-default.webp&w=3840&q=75 3840w"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
              />
            </div>
            <div
              class="flex justify-between pt-5"
            >
              <div>
                <p
                  class="text-primary dark:text-white text-extra font-bold leading-large text-large"
                >
                  Mesh Gradients
                </p>
                <p
                  class="text-secondary dark:text-secondary text-small font-medium"
                >
                  By User 1
                </p>
              </div>
              <div
                class="flex items-center justify-center h-auto w-max [&>span]:w-[27px] [&>span]:h-[27px] [&>span]:ring-offset-0"
                role="group"
              >
                <span
                  class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-10 h-10 text-tiny bg-default text-default-foreground rounded-full ring-2 ring-offset-2 ring-offset-background dark:ring-offset-background-dark data-[hover=true]:-translate-x-3 rtl:data-[hover=true]:translate-x-3 transition-transform data-[focus-visible=true]:-translate-x-3 rtl:data-[focus-visible=true]:translate-x-3 ring-default ms-0"
                  tabindex="-1"
                >
                  <img
                    alt="Adela Parkson"
                    class="flex object-cover w-full h-full transition-opacity !duration-500 opacity-0 data-[loaded=true]:opacity-100"
                    data-nimg="1"
                    decoding="async"
                    height="27"
                    loading="lazy"
                    src="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=64&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=64&q=75 2x"
                    style="color: transparent;"
                    width="27"
                  />
                </span>
                <span
                  class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-10 h-10 text-tiny bg-default text-default-foreground rounded-full ring-2 ring-offset-2 ring-offset-background dark:ring-offset-background-dark data-[hover=true]:-translate-x-3 rtl:data-[hover=true]:translate-x-3 transition-transform data-[focus-visible=true]:-translate-x-3 rtl:data-[focus-visible=true]:translate-x-3 ring-default -ms-2"
                  tabindex="-1"
                >
                  <img
                    alt="user2"
                    class="flex object-cover w-full h-full transition-opacity !duration-500 opacity-0 data-[loaded=true]:opacity-100"
                    data-nimg="1"
                    decoding="async"
                    height="27"
                    loading="lazy"
                    src="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=64&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=64&q=75 2x"
                    style="color: transparent;"
                    width="27"
                  />
                </span>
                <span
                  class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-10 h-10 text-tiny bg-default text-default-foreground rounded-full ring-2 ring-offset-2 ring-offset-background dark:ring-offset-background-dark data-[hover=true]:-translate-x-3 rtl:data-[hover=true]:translate-x-3 transition-transform data-[focus-visible=true]:-translate-x-3 rtl:data-[focus-visible=true]:translate-x-3 ring-default -ms-2"
                  tabindex="-1"
                >
                  <img
                    alt="user3"
                    class="flex object-cover w-full h-full transition-opacity !duration-500 opacity-0 data-[loaded=true]:opacity-100"
                    data-nimg="1"
                    decoding="async"
                    height="27"
                    loading="lazy"
                    src="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=64&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdn5vw6fjp%2Fimage%2Fupload%2Fv1726630608%2Fimage_de1e29b0b7.png&w=64&q=75 2x"
                    style="color: transparent;"
                    width="27"
                  />
                </span>
                <span
                  class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-10 h-10 text-tiny bg-default text-default-foreground rounded-full ring-2 ring-offset-2 ring-offset-background dark:ring-offset-background-dark -ms-2 data-[hover=true]:-translate-x-3 rtl:data-[hover=true]:translate-x-3 transition-transform data-[focus-visible=true]:-translate-x-3 rtl:data-[focus-visible=true]:translate-x-3 ring-default hover:-translate-x-0"
                  tabindex="-1"
                >
                  <span
                    aria-label="+1"
                    class="absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 font-normal text-center text-inherit"
                    role="img"
                  >
                    +1
                  </span>
                </span>
              </div>
            </div>
            <p
              class="text-primary dark:text-white text-small text-blue-450 font-bold pt-[25px]"
            >
              Current Bid: 1.23 USD
            </p>
          </div>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
